hal/src/thumbv7em/pukcc/c_abi.rs

```diff
@@ -0,0 +1,822 @@
+    pub u2Status: PukclReturnCode,
+    pub __Padding0: u2,
+    pub __Padding1: u4,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Default)]
+pub struct PukclReturnCode(pub u2);
+
+impl core::fmt::Debug for PukclReturnCode {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        super::PukclReturnCode::from(*self).fmt(f)
+    }
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union PukclParamsUnion {
+    pub ClearFlags: ClearFlags,
+    pub Comp: Comp,
+    pub CondCopy: CondCopy,
+    pub CRT: CRT,
+    pub Div: Div,
+    pub ExpMod: ExpMod,
+    pub FastCopy: FastCopy,
+    pub Fill: Fill,
+    pub Fmult: Fmult,
+    pub GCD: GCD,
+    pub PrimeGen: PrimeGen,
+    pub RedMod: RedMod,
+    pub Rng: Rng,
+    pub SelfTest: SelfTest,
+    pub Smult: Smult,
+    pub Square: Square,
+    pub Swap: Swap,
+
+    /* ECC on Prime Field */
+    pub ZpEccAddFast: ZpEccAddFast,
+    pub ZpEccDblFast: ZpEccDblFast,
+    pub ZpEccAddSubFast: ZpEccAddSubFast,
+    pub ZpEccMulFast: ZpEccMulFast,
+    pub ZpEcDsaGenerateFast: ZpEcDsaGenerateFast,
+    pub ZpEcDsaVerifyFast: ZpEcDsaVerifyFast,
+    pub ZpEcDsaQuickVerify: ZpEcDsaQuickVerify,
+    pub ZpEccQuickDualMulFast: ZpEccQuickDualMulFast,
+    pub ZpEcConvProjToAffine: ZpEcConvProjToAffine,
+    pub ZpEcConvAffineToProjective: ZpEcConvAffineToProjective,
+    pub ZpEcRandomiseCoordinate: ZpEcRandomiseCoordinate,
+    pub ZpEcPointIsOnCurve: ZpEcPointIsOnCurve,
+
+    /* ECC on Binary Field */
+    pub GF2NEccAddFast: GF2NEccAddFast,
+    pub GF2NEccDblFast: GF2NEccDblFast,
+    pub GF2NEccMulFast: GF2NEccMulFast,
+    pub GF2NEcDsaGenerateFast: GF2NEcDsaGenerateFast,
+    pub GF2NEcDsaVerifyFast: GF2NEcDsaVerifyFast,
+    pub GF2NEcConvProjToAffine: GF2NEcConvProjToAffine,
+    pub GF2NEcConvAffineToProjective: GF2NEcConvAffineToProjective,
+    pub GF2NEcRandomiseCoordinate: GF2NEcRandomiseCoordinate,
+    pub GF2NEcPointIsOnCurve: GF2NEcPointIsOnCurve,
+}
+
+impl core::default::Default for PukclParamsUnion {
+    fn default() -> Self {
+        Self {
+            ClearFlags: Default::default(),
+        }
+    }
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Default)]
+pub struct PukclParams {
+    pub header: PukclHeader,
+    pub params: PukclParamsUnion,
+}
+
+impl core::fmt::Debug for PukclParams {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("PukclParams")
+            .field("header", &self.header)
+            .field(
+                "params",
+                match debug_union_matcher(self) {
+                    Ok(field) => field,
+                    Err(_) => &"invalid service",
+                },
+            )
+            .finish()
+    }
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Default, Debug)]
+pub struct ZpEccQuickDualMulFast {
+    pub pu1ModCnsBase: pu1,
+    pub pu1PointABase: pu1,
+    pub pu1PointBBase: pu1,
+    pub pu1KABBase: pu1,
+    pub pu1AWorkBase: pu1,
+    // pu1       pu1ABase,
```

Comment:
Not needed?

Reply:
Removed all comments and weird whitespace from `c_abi` module